### PR TITLE
feat(jans-cedarling): Support JSON logic for principal boolean operations

### LINKED ISSUE #10590: feat(jans-cedarling): Support JSON logic for principal boolean operations
## **Is your feature request related to a problem? Please describe.**

The current `CEDARLING_USER_WORKLOAD_BOOLEAN_OPERATION` bootstrap property is restrictive, allowing only basic `AND` or `OR` operations to combine the results of multiple principals. This limitation prevents the implementation of more complex decision-making logic or the inclusion of additional custom principals.

## **Describe the solution you'd like**

Replace the `CEDARLING_USER_WORKLOAD_BOOLEAN_OPERATION` property with a new `CEDARLING_PRINCIPAL_BOOLEAN_OPERATION` property that accepts a [JSON Logic](https://jsonlogic.com/) expression. This enhancement would enable users to define more flexible and complex logic for combining principal results.

### Examples

1. **Current `AND` in JSON Logic:**

```json
{
    "and" : [
        {"==": [{"var": "Jans::Workload"}, "ALLOW"]},
        {"==": [{"var": "Jans::User"}, "ALLOW"]}
    ] 
}
```

2. **Current `OR `in JSON Logic:**

```json
{
    "and" : [
        {"==": [{"var": "Jans::Workload"}, "ALLOW"]},
        {"==": [{"var": "Jans::User"}, "ALLOW"]}
    ] 
}
```

3. **Complex Decision Tree Example:** An `ALLOW` decision is returned only if either the `Jans::Workload` or `Jans::User` principal is allowed, and a custom principal (`Another::Custom`) is also allowed:

```json
{
    "and" : [
        {"or": [
            {"==": [{"var": "Jans::Workload"}, "ALLOW"]},
            {"==": [{"var": "Jans::User"}, "ALLOW"]}
        ]},
        {"==": [{"var": "Another::Custom"}, "ALLOW"]}
    ] 
}
```

### JSON Logic for Rust

There are multiple crates that we can try and use to parse the JSON logic:
 
- [datalogic-rs](https://crates.io/crates/datalogic-rs): around 4k recent downloads
- [jsonlogic-rs](https://crates.io/crates/jsonlogic-rs): around 3k recent downloads
- [jsonlogic](https://crates.io/crates/jsonlogic): around 1.5k recent downloads

It also seems like `jsonlogic` isn't really being maintained anymore.

## **Describe alternatives you've considered**
N/A

## **Additional context**

To explore and test JSON Logic expressions, visit: https://jsonlogic.com/play.html

**Example Rule:**

```json
{
    "and" : [
        {"==": [{"var": "Jans::Workload"}, "ALLOW" ] },
        {"==": [{"var": "Jans::User"}, "ALLOW" ] }
    ] 
}
```

Example data:

```json
{
    "Jans::User" : "ALLOW",
    "Jans::Workload" : "ALLOW"
}
```

### PR DESCRIPTION
**Pull request by olehbozhok**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description
add `CEDARLING_USER_WORKLOAD_BOOLEAN_OPERATION ` that contains json logic rules
#### Target issue
  
  [link](https://github.com/JanssenProject/jans/issues/10590)
  
  
closes #10590

#### Implementation Details
  
added `CEDARLING_USER_WORKLOAD_BOOLEAN_OPERATION ` that contains json logic rules

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [x] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [ ] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

**Review by rmarinn — DISMISSED** [review]

**Review by nynymike — COMMENTED** [review]
I think you should add more tests. I feel like you only tested the one case you are interested in. You should try a whole bunch of different use cases. Also, you need to test negative--make sure access is denied in expected use cases.

File: docs/cedarling/cedarling-properties.md [hunk 1]
```diff
@@ -77,7 +77,22 @@ All other fields are optional and can be omitted. If a field is not provided, Ce
 
 ## User-Workload Boolean Operation
 
-The `CEDARLING_USER_WORKLOAD_BOOLEAN_OPERATION` property specifies what boolean operation to use for the `USER` and `WORKLOAD` when making authz (authorization) decisions.
+The `CEDARLING_PRINCIPAL_BOOLEAN_OPERATION` property specifies what boolean operation to use for the `USER` and `WORKLOAD` when making authz (authorization) decisions.
+
+Make sure that you use correct `var` name for `principal` types.
+
+Use [JsonLogic](https://jsonlogic.com/).
```
Comment (SafinWasi) [inline docs/cedarling/cedarling-properties.md]: I think we should provide an explanation on what exactly the default payload means and provide some examples. Not everyone is familiar with jsonlogic, and we should provide an explanation instead of expecting developers to learn a format with no formal specification.

**Review by SafinWasi — CHANGES_REQUESTED** [review]
Python components look good to me, and works on my test setup. However, JsonLogic is not a well known format and needs more explanation on the default value of `CEDARLING_PRINCIPAL_BOOLEAN_OPERATION`, and maybe some examples.

File: docs/cedarling/cedarling-properties.md (hunk 1 above)
Comment (olehbozhok) [reply docs/cedarling/cedarling-properties.md]: Updated

**Comment by olehbozhok** [comment]
Added test cases against jsonlogic and updated documentation

File: docs/cedarling/cedarling-properties.md [hunk 2]
```diff
@@ -350,7 +469,7 @@ CEDARLING_JWT_SIGNATURE_ALGORITHMS_SUPPORTED:
     - HS256
     - RS256
 
-CEDARLING_ID_TOKEN_TRUST_MODE: strict
+**CEDARLING_ID_TOKEN_TRUST_MODE**: strict
```
Comment (rmarinn) [inline docs/cedarling/cedarling-properties.md]: i think adding the asterisks here was not intended `** ... **`.

It looks like this:
image

Comment (olehbozhok) [reply docs/cedarling/cedarling-properties.md]: Thanks, resolved
